### PR TITLE
feat(emails): get email service to work with mail helper

### PR DESCRIPTION
**Pull request by brizental**
These changes were made because of: https://github.com/mozilla/fxa-email-service/pull/122

The major changes are:

- If we are running locally it will not use SES by default, it will send emails to `mail_helper.js`.
- I added a log for `mail_helper.js` that says which port the SMTP server is listening to.
- I set the provider for local development to default to `smtp` and prod is `ses`. Also this makes it easier if ever we want to send some emails to sendgrid or any other provider, we can just add an optional argument to `emailService.sendMail` that overwrites whatever came from config.

This means that development workflow would be exactly the same if you are using Nodemailer or fxa-email-service. 

Finally, I was thinking. Should we add fxa-email-service to fxa-local-dev? Makes sense to me that we add it.

r? @mozilla/fxa-devs

### CODE REVIEW

File: lib/senders/email_service.js [hunk 1]
```diff
@@ -25,7 +25,8 @@ module.exports = (config) => {
         body: {
           text: emailConfig.text,
           html: emailConfig.html
-        }
+        },
+        provider: config.emailService.provider
```
[the remark this answers is not in the document]
Comment (philbooth) [reply lib/senders/email_service.js]: (longer-term we will want to set this on *some* requests, but it will need a more intricate control mechanism than a simple pref; for instance, one use case we talked about was the possibility of selectively diverting certain users who are experiencing deliverability problems via Sendgrid or whatever)

**Comment by philbooth** [comment]
> Should we add fxa-email-service to fxa-local-dev?

Yep, definitely!

File: lib/senders/email_service.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (philbooth) [reply lib/senders/email_service.js]: Alternatively, I'd also be happy with us always setting `provider` like you do here, then making it non-optional and removing the setting from `fxa-email-service`. Either is good, I just don't want there to be two settings is all.

[the remark this answers is not in the document]
Comment (brizental) [reply lib/senders/email_service.js]: > longer-term we will want to set this on some requests, but it will need a more intricate control mechanism than a simple pref

In this case I thought we could just add an optional argument to the `emailService.sendMail` function that would overwrite the config.

[the remark this answers is not in the document]
Comment (vladikoff) [reply lib/senders/email_service.js]: > selectively diverting certain users who are experiencing deliverability problems

I think this is key 🗝 !

File: test/mail_helper.js [hunk 2]
```diff
@@ -99,7 +99,15 @@ module.exports = (printLogs) => {
         req.accept()
       }
     )
-    smtp.listen(config.smtp.port, config.smtp.host)
+
+    smtp.listen(config.smtp.port, function(err) {
```
Comment (vladikoff) [inline test/mail_helper.js]: Let's extract this into a different PR.
